### ISSUE TITLE
Enable torch.nn.modules.* typechecks during CI

### ISSUE THREAD

**Issue by guilhermeleobas** [pytorch/pytorch#45771]
`TestTypeHints.test_run_mypy` should pass with `[mypy-torch.nn.modules.(batchnorm/instancenorm/linear/sparse)]` deleted from mypy.ini

cc @mruberry @VitalyFedyunin @ezyang @malfet @rgommers @xuzhao9 @gramster
